trajopt_ifopt/src/absolute_cost.cpp

```diff
@@ -0,0 +1,83 @@
+/**
+ * @file absolute_cost.cpp
+ * @brief The absolute cost. Converts a constraint into a cost.
+ *
+ * @author Levi Armstrong
+ * @date May 20, 20201
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2021, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <trajopt_ifopt/costs/absolute_cost.h>
+#include <trajopt_ifopt/utils/ifopt_utils.h>
+
+TRAJOPT_IGNORE_WARNINGS_PUSH
+#include <console_bridge/console.h>
+TRAJOPT_IGNORE_WARNINGS_POP
+
+namespace trajopt
+{
+AbsoluteCost::AbsoluteCost(ifopt::ConstraintSet::Ptr constraint)
+  : AbsoluteCost(std::move(constraint), Eigen::VectorXd::Ones(constraint->GetRows()))
+{
+}
+
+AbsoluteCost::AbsoluteCost(ifopt::ConstraintSet::Ptr constraint, const Eigen::Ref<const Eigen::VectorXd>& weights)
+  : CostTerm(constraint->GetName() + "_absolute_cost")
+  , constraint_(std::move(constraint))
+  , n_constraints_(constraint_->GetRows())
+  , weights_(weights.cwiseAbs())  // must be positive
```

Comment:
Ideally I think we should throw an error if weights is the wrong size. Otherwise I think Eigen will throw in FillJacobianBlock with a probably more cryptic error